spp_programs/wizard/create_program_wizard.py

```diff
@@ -34,10 +35,29 @@ def on_admin_area_ids_change(self):
         if self.admin_area_ids:
             area_ids = self.admin_area_ids.ids
             domain.append(("area_id", "in", tuple(area_ids)))
-            eligibility_domain = str(domain)
+        eligibility_domain = str(self._insert_domain_operator(domain))
 
         self.eligibility_domain = eligibility_domain
 
+    def _insert_domain_operator(self, domain):
+        """Insert operator to the domain"""
+        operator_used = "&"
+        operator = {
+            "&": AND,
+            "|": OR,
+        }
+        new_domain = []
+        if domain:
+            # Check and get the operator used in the domain
+            if domain[0] in ["&", "|"]:
+                operator_used = domain[0]
+
+            domain = list(filter(lambda a: a != operator_used, domain))
+            for dom in domain:
+                new_domain = operator[operator_used]([new_domain, [dom]])
+
+        return new_domain
```

Comment:
```suggestion
        if not domain:
            return domain
        operator_used = AND
        if domain[0] == "|":
            operator_used = OR
        new_domain = []
        domain = list(filter(lambda a: a not in ["&", "|", "!"], domain))
        for dom in domain:
            new_domain = operator_used([new_domain, [dom]])

        return new_domain
```
@reichie020212 can this code look like this?

Reply:
@nhatnm0612 looks better, I'll change it to this. Thanks.